### PR TITLE
Implement save offer functionality

### PR DESCRIPTION
**Pull request by eliseydah**
Add price calculation for the offer.

Add articles filtering and pre-fill.

Change Offer Page layout.

Minor improvement.

**Important:**

Final price calculation for offer will be added in the next pull request.

### CODE REVIEW

File: frontend/src/views/OfferView.vue [hunk 1]
```diff
@@ -1,60 +1,123 @@
...
+const items = [
+  {
+    label: "Add note",
+    command: () => {
+      addOfferItem(ArticleKind.heading);
+      toast.add({ severity: "success", detail: "Note added", life: 3000 });
+    },
+  },
+  {
+    label: "Add position",
```
Comment (fr-ser) [inline frontend/src/views/OfferView.vue:66]:
```suggestion
    label: "Position hinzufügen",
```

File: frontend/src/views/OfferView.vue [hunk 2]
```diff
@@ -1,60 +1,123 @@
...
+    command: () => {
+      addOfferItem(ArticleKind.item),
+        toast.add({ severity: "success", detail: "Position added", life: 3000 });
```
Comment (fr-ser) [inline frontend/src/views/OfferView.vue:69]:
```suggestion
        toast.add({ severity: "success", detail: "Position hinzugefügt", life: 3000 });
```

File: frontend/src/views/OfferView.vue [hunk 3]
```diff
@@ -1,60 +1,123 @@
...
   offerItemsArray.value.push({ id: itemCount++, type: type });
 }
 
+const toast = useToast();
+const items = [
+  {
+    label: "Add note",
+    command: () => {
+      addOfferItem(ArticleKind.heading);
+      toast.add({ severity: "success", detail: "Note added", life: 3000 });
```
Comment (fr-ser) [inline frontend/src/views/OfferView.vue:62]:
```suggestion
      toast.add({ severity: "success", detail: "Hinweis hinzugefügt", life: 3000 });
```

File: frontend/src/views/OfferView.vue [hunk 4]
```diff
@@ -1,60 +1,123 @@
...
+  if (offerDate.value) {
+    offerInfo.value.offered_at = format(offerDate.value, "yyyy-MM-dd");
+  } else {
+    offerInfo.value.offered_at = "";
+  }
+});
+
+watch(validityDate, () => {
+  if (validityDate.value) {
+    offerInfo.value.offer_valid_until = format(validityDate.value, "yyyy-MM-dd");
```
Comment (fr-ser) [inline frontend/src/views/OfferView.vue:98]: Is format the same as "formatIsoDateString" here?

File: frontend/src/views/OfferView.vue [hunk 5]
```diff
@@ -127,29 +190,13 @@ onMounted(async () => {
...
-            outlined
-            size="small"
-          ></Button>
+          <div class="font-bold">Summe:</div>
+          <div class="flex flex-row gap-10">
+            <span>Netto: </span>
+            <span>USt: </span>
+            <span>Brutto: </span>
+          </div>
+          <SplitButton label="Add" :model="items" :class="'w-full'" />
```
Comment (fr-ser) [inline frontend/src/views/OfferView.vue:199]:
```suggestion
          <SplitButton label="Hinzufügen" :model="items" :class="'w-full'" />
```

File: frontend/src/views/OfferView.vue (hunk 4 above)
Comment (eliseydah) [reply frontend/src/views/OfferView.vue:93]: we have another format and I added helper function for the formatting

**Pull request merged**